ttax/ops.py

```diff
@@ -368,3 +407,28 @@ def _mul_by_scalar(tt, c):
     return TTMatrix(cores)
   else:
     return TT(cores)
+
+
+def transpose(tt):
+  """Transpose a TT-matrix or a batch of TT-matrices.
+
+  :type tt:   `TT-Matrix` object containing TT-matrix or batch of TT-matrices
+  :param tt:  TT-matrix or batch of TT-matrices
+  :return:    `TT-Matrix` object containing TT-matrix or batch of TT-matrices
+  :rtype:     `TT-Matrix` object containing TT-matrix or batch of TT-matrices
+  :raises [ValueError]: if the argument is not a TT-matrix
+  """
+  if not isinstance(tt, TTMatrix) or not tt.is_tt_matrix:
+    raise ValueError('The argument should be a TT-matrix.')
+
+  if tt.num_batch_dims == 0:
+    transposed_tt_cores = [
+      jnp.transpose(tt.tt_cores[core_idx], (0, 2, 1, 3)) for core_idx in range(tt.ndim)
+    ]
+  else:
+    # Batch of TT-matrices
+    transposed_tt_cores = [
+      jnp.transpose(tt.tt_cores[core_idx], (0, 1, 3, 2, 4)) for core_idx in range(tt.ndim)
```

Comment:
In TTAX we support multidimensional batches (and for a good but a bit obscure reason, let me know if you  want details). So if it's a batch you don't know if it adds a single dimension or multiple.

But on the bright side, we have the tt_vmap decorator. So you can basically do this
```
@tt_vmap
def transpose(...):
  if not tt.is_tt_matrix: raise...
  transposed_tt_cores = [
      jnp.transpose(tt.tt_cores[core_idx], (0, 2, 1, 3)) for core_idx in range(tt.ndim)
    ]
    return TTMatrix(transposed_tt_cores)